README.md

```diff
@@ -1,73 +1,156 @@
-## FAQ
-
-> Where should I send back the result when I'm done?
-
-Fork this repo and send us a pull request when you think it's ready for review. You don't have to finish everything prior and you can continue to work on it. We don't have a deadline for the task.
-
-> What if I have a question?
-
-Create a new issue in the repo and we will get back to you shortly.
+# Test Backend from WiredCraft
+
+## Overview
+
+This is the test-backend project from Wiredcraft for a general `user` "microservice". 
+
+---
+
+## Quick Notes regard to the reqs
+Supported APIs are:
+ - GET /users
+ - GET /users/:username 
+ - GET /users/:username/follower
+ - POST /users
+ ```json
+ // Create a user
+  // sample body and res
+ {
+    "name": "username",
+    "password": "password",
+    "address": "1234 Main St",
+    "description": "test description"
+ }
+ {
+    "errCode": -1,
+    "data": 1
+ }
+  // data: the created userId, actual return value may vary from business needs
+ ```
+ - POST /users/delete 
+ ```json
+ // Delete a user by username,
+  // sample body and res
+ {
+   "username": "username"
+ }
+ {
+   "errCode": -1
+ }
+ ```
+ - POST /users/follow
+ ```json
+   // follow a user
+   // sample body and res
+ {
+   "username": "username",
+   "follower": "follower"
+ }
+ {
+    "errCode": 10008,
+    "msg": "can not follow or get followed by void"
+}
+ ```
+ - POST /users/unfollow
+ ```json
+ // unfollow a user
+ // sample body and res
+ {
+   "username": "435",
+   "follower": "follo123wer"
+ }
+{
+    "errCode": 10010,
+    "msg": "given following relation does not exist"
+}
+ ```
+ - PATCH /users/info
+ ```json
+ // update user info by username
+ // sample body and res
+{
+    "name": "follower",
+    "address": "new address"
+}
+ {
+    "errCode": -1,
+    "data": {
+        "id": 47,
+        "name": "follower",
+        "dateOfBirth": null,
+        "address": "new address",
+        "description": "test description"
+    }
+}
+ ```
+ - PATCH /users/password
+```json
+// update user pwd by username
+// sample body and res
+{
+    "name": "follower",
+    "oldPassword": "password",
+    "newPassword": "newpwd"
+}
+{
+   "errCode": -1
+}
+```
+#### Responses example,
+```json
+// errCode: Shared property for all res, -1 on success, positive pre-defined code for error
+// data: Optional field for success response, if data needed
+// message: Err message if any, for Error Response
+{
+   "errCode": "number", 
+   "data": {}, 
+   "message": "string" 
+}
+```
+~~You might have noticed that the `api routes` are lacking auth middleware that protects those who needs a guard, it's simply because auth/login is not implemented, does not mean that those APIs do not need auth protection.~~
+
+---
+
+## Dependencies
+
+As I have not tested for version compatibilities, assume the latest version of following would work. 
+ - typescript/ts-node
+ - docker(for mysql and redis)
+ - jest/ts-jest
+ - pm2
+---
+## How
+   - docker run -d -p 6379:6379 --name redis1 redis

```

Comment:
why not also put redis in docker-compose file?

Reply:
could added a " redis: "  section in docker compose. 
I didn't plan to integrate this at first place, it was just a last minute 'lazy one liner'